### ISSUE TITLE
Better error messages for missing opcodes

### ISSUE THREAD

**Issue by JesusFreke** [JesusFreke/smali#500]
When assembling a smali file from the O preview with invoke-polymorphic instructions, if smali is invoked without specifying an api, the default set of opcodes for api 15 get used, which doesn't contain the invoke-polymorphic opcode. When smali encounters one of these opcodes, the error message is.. unhelpful.

    boot.oat_deodex/java/lang/invoke/Transformers$FilterReturnValue.smali[180,4] null

**Comment by Furniel** [comment]
Getting same errors during Pixel XL android O boot.oat(core-oj) deodexing. Api level was specified.
```
tmp\java\lang\invoke\Transformers$CatchException.smali[167,4] null
tmp\java\lang\invoke\Transformers$Construct.smali[238,4] null
tmp\java\lang\invoke\Transformers$Collector.smali[277,4] null
tmp\java\lang\invoke\Transformers$FilterArguments.smali[228,4] null
tmp\java\lang\invoke\Transformers$FoldArguments.smali[211,4] null
tmp\java\lang\invoke\Transformers$GuardWithTest.smali[112,4] null
tmp\java\lang\invoke\Transformers$CollectArguments.smali[224,4] null
tmp\java\lang\invoke\Transformers$InsertArguments.smali[413,4] null
tmp\java\lang\invoke\Transformers$BindTo.smali[113,4] null
tmp\java\lang\invoke\Transformers$DropArguments.smali[145,4] null
tmp\java\lang\invoke\Transformers$FilterReturnValue.smali[180,4] null
```

Commands that were used:
```
java -jar baksmali.jar x -a 25 -c framework\arm64\boot.oat -d framework\arm64 framework\arm64\boot.oat/system/framework/core-oj.jar -o tmp
java -jar smali.jar a -a 25 -o classes.dex tmp
```
And one more error: Smali shows this errors, don't creates dex file but exits with code 0.

**Comment by JesusFreke** [comment]
Ah, right, I need to update the oat <-> api version map. For now, use -a 26 instead.

**Comment by zengfu94** [comment]
@JesusFreke even using -a 26 I get the same errors as the user above.

**Comment by Furniel** [comment]
It should work. With -a 26 everything works fine, during tests i successfully deodexed all released android O images(Nexus 5X, Nexus 6P, Nexus Player, Pixel C, Pixel, Pixel XL).

**Comment by JesusFreke** [comment]
@zengfu94 don't forget -a 26 to smali ;)

@Furniel nice! Thanks for confirming that it's all working for you.

**Comment by zengfu94** [comment]
@Furniel what tool did you use? I will try again

**Comment by Furniel** [comment]
@JesusFreke Sorry for offtop.
@zengfu94 I'm using [own program](https://github.com/Furniel/Apk-Changer).

**Comment by Aaahh** [comment]
@Furniel offtop but can you send me a system dump of Android O!

**Comment by JesusFreke** [comment]
@Aaahh see https://developer.android.com/preview/index.html

**Comment by Aaahh** [comment]
@JesusFreke I can't extract the new images as the system image is a sparse image with a new format

**Comment by JesusFreke** [comment]
The standard simg2img tool that I've always used worked fine.

**Comment by Aaahh** [comment]
@JesusFreke yeah I've used it before but with these new images there's no transfer_list and it fails

**Comment by JesusFreke** [comment]
Worked fine for me :). Are you using the simg2img tool from AOSP?

**Comment by Aaahh** [comment]
@JesusFreke Ah I knew I was forgetting something. No I was using some script off GitHub. Whoops my bad, thank you
